test/opt/test_pass_manager.cpp

```diff
@@ -54,6 +56,17 @@ TEST(PassManager, Interface) {
   EXPECT_STREQ("strip-debug", manager.GetPass(0)->name());
   EXPECT_STREQ("null", manager.GetPass(1)->name());
   EXPECT_STREQ("strip-debug", manager.GetPass(2)->name());
+
+  manager.AddPass<opt::NullPass>(1u);
+  manager.AddPass<opt::NullPass>("null pass args");
+  manager.AddPass<opt::NullPass>(std::initializer_list<int>{1, 2});
+  EXPECT_EQ(6u, manager.NumPasses());
+  EXPECT_STREQ("strip-debug", manager.GetPass(0)->name());
+  EXPECT_STREQ("null", manager.GetPass(1)->name());
+  EXPECT_STREQ("strip-debug", manager.GetPass(2)->name());
+  EXPECT_STREQ("null", manager.GetPass(3)->name());
```

Comment:
Additionally you can do `static_cast` to the real type and access some of the data passed in via constructor.